feature: Add rviz_web in robot_blockly

lianwenbo opened the pull request:
1)在pywebserver启动的时候，用subprocess同时启动rviz需要的rosnode和roslaunch
2)在blockly.html的界面上，用iframe嵌套了html，用以展示rviz，在测试上,interactivemarker
3)需要注意的相关配置在pyserver上，用commands指定了需要启动的命令行，目前默认是interactivemarker，需要配置
4)在blockly.html上，目前使用的是interactivemarker的html，如果需要用urdf.html的话，在配置的时候要改正
5)在urdf.html与interactivemarker的ros的url，这个url应该是服务器的ip，如果从外网连接的话，需要将其替换成外网的ip

lazyparser commented on frontend/pages/rviz/interactivemarkers.html:
```diff
@@ -0,0 +1,55 @@
+    // Connect to ROS.
+    var ros = new ROSLIB.Ros({
+      // set according to the server ip
+      url : 'ws://192.168.56.101:9090'
```
这里是需要从主页面获取

lianwenbo replied:
这个怎么从主页面获得，不了解诶，但是一定是要主机的对外ip

lazyparser replied:
后续的实现是配置的时候配置好一个ROSCore的IP地址，所有的组件都可以查询到这个IP地址，进行连接；或者直接做一个负责跟ROSCore进行连接的模块。

lazyparser commented on frontend/pages/blockly.html:
```diff
@@ -207,7 +207,12 @@
+		    <div class="col-sm-6" id="test" style="height:100vh;">
+			<h1>Rviz Web</h1>
+			<iframe width="520px" height="490px" src="./rviz/interactivemarkers.html"></iframe>
+			<!--iframe width="520px" height="490px" src="./rviz/urdf.html"></iframe-->
```
这行可以删掉了

lazyparser commented on frontend/pages/rviz/interactivemarkers.html:
```diff
@@ -0,0 +1,55 @@
+<head>
+<meta charset="utf-8" />
+
+<script src="http://cdn.robotwebtools.org/threejs/current/three.js"></script>
```
本地化。不要用cdn。我们的环境可能连不到外网。

lianwenbo replied:
嗯，这个可以试试

lazyparser commented on frontend/pages/rviz/interactivemarkers.html:
```diff
@@ -0,0 +1,55 @@
+
+<script src="http://cdn.robotwebtools.org/threejs/current/three.js"></script>
+<script src="http://cdn.robotwebtools.org/EventEmitter2/current/eventemitter2.min.js"></script>
+<script src="http://cdn.robotwebtools.org/roslibjs/current/roslib.js"></script>
```
这几个文件都是。

lazyparser commented on frontend/pages/rviz/interactivemarkers.html:
```diff
@@ -0,0 +1,55 @@
+    var viewer = new ROS3D.Viewer({
+      divID : 'markers',
+      width : 800,
+      height : 600,
```
这里形状写了之后，后续可以通过某种方式进行修改或者扩展么？
例如，在网页中“全屏”或者“半屏”展示。

lazyparser commented on frontend/pages/rviz/interactivemarkers.html:
```diff
@@ -0,0 +1,55 @@
+      alpha : 0.5
+    });
+
+    // Setup a client to listen to TFs.
```
这里是不是跟具体的topic硬绑定了。
问一下 ck

lazyparser commented on frontend/pages/rviz/urdf.html:
```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
```
这个文件如果没有使用，删掉？

lianwenbo replied:
interactivemarker.html是我在虚拟机上通过的，但是urdf似乎有些问题，这个文件我想到你们的服务器上测试一下

lazyparser commented on scripts/pywebserver.py:
```diff
@@ -3,16 +3,33 @@
+	    'roslaunch rosbridge_server rosbridge_websocket.launch']
+
+'''
+commands = ['roslaunch pr2_description upload_pr2.launch',\
```
这段是干什么，是不是注释忘记删除了？

lazyparser commented on scripts/pywebserver.py:
```diff
@@ -3,16 +3,33 @@
+	    'roslaunch rosbridge_server rosbridge_websocket.launch']
+'''
+
+for command in commands:
```
作为一开始调试方便直接调用是可以的。比较好的方式是直接使用ros的launch文件来进行。

lianwenbo replied:
这个后续的修改再作处理

lazyparser replied:
没问题的，加分项。

lazyparser reviewed (commented):
Nice Try :-)

lianwenbo committed:
feature: Add rviz_web in robot_blockly

1)在pywebserver启动的时候，用subprocess同时启动rviz需要的rosnode和roslaunch
2)在blockly.html的界面上，用iframe嵌套了html，用以展示rviz，在测试上,interactivemarker
3)需要注意的相关配置在pyserver上，用commands指定了需要启动的命令行，目前默认是interactivemarker，需要配置
4)在blockly.html上，目前使用的是interactivemarker的html，如果需要用urdf.html的话，在配置的时候要改正
5)在urdf.html与interactivemarker的ros的url，这个url应该是服务器的ip，如果从外网连接的话，需要将其替换成外网的ip

lazyparser commented:
由于  scripts/pywebserver.py  已经重写成 python2 版本了，需要重新 rebase 一下这个PR。

lazyparser reviewed (approved):
LGTM
